### PR TITLE
[11.0][IMP] date_range: adapt to multicompany

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
This PR adapts the `date_range` module to a multicompany environment.

It also provides tests for this improvement and migration files.

I hope all you like it.

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@etobella @jbeficent

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Tested. Works like a charm!

File: date_range/models/date_range_type.py [hunk 1]
```diff
@@ -25,3 +27,19 @@ def _default_company(self):
...
         ('date_range_type_uniq', 'unique (name,company_id)',
          'A date range type must be unique per company !')]
+
+    @api.constrains('company_id')
+    def _check_company_id(self):
+        if not self.env.context.get('bypass_company_validation', False):
+            for rec in self:
+                if not rec.company_id:
+                    continue
+                field = self.env['date.range'].search(
```
Comment (lmignon) [inline date_range/models/date_range_type.py]: @mreficent  you could maybe avoid a search into a loop  by using a `read_group` (not bloking)

**Review by lmignon — APPROVED** [review]
Thank you for this improvement. (Code review only)

File: date_range/models/date_range_type.py (hunk 1 above)
Comment (feketemihai) [reply date_range/models/date_range_type.py]: Probably it's better to have a one2many to date_range and use mapped on company_id.

Comment (MiquelRForgeFlow) [reply date_range/models/date_range_type.py]: Yeah, I like this one2many approach. I will move this way.

**Comment by pedrobaeza** [comment]
@jbeficent please wait until CIs are green before merging.

**Pull request merged**